packages/sdk-react/src/components/receivables/InvoiceDetails/ExistingInvoiceDetails/components/InvoiceCancelModal.tsx

```diff
@@ -0,0 +1,95 @@
+import { useRootElements } from '@/core/context/RootElementsProvider';
+import {
+  useCancelReceivableById,
+  useReceivableById,
+} from '@/core/queries/useReceivables';
+import { t } from '@lingui/macro';
+import { useLingui } from '@lingui/react';
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Divider,
+  Skeleton,
+} from '@mui/material';
+
+interface InvoiceCancelModalProps {
+  invoiceId: string;
+  open: boolean;
+  onClose: () => void;
+}
+
+export const InvoiceCancelModal = ({
+  invoiceId,
+  open,
+  onClose,
+}: InvoiceCancelModalProps) => {
+  const { i18n } = useLingui();
+  const { root } = useRootElements();
+  const { data: receivable, isLoading: isReceivableLoading } =
+    useReceivableById(invoiceId);
+  const cancelReceivableMutation = useCancelReceivableById(invoiceId);
+
+  const creditNoteIds =
+    receivable?.type === 'invoice'
+      ? receivable.related_documents.credit_note_ids
+      : undefined;
+
+  const hasNoCreditNotes = !creditNoteIds?.length;
+
+  const modalTitle = !receivable ? (
+    <Skeleton />
+  ) : receivable.document_id ? (
+    t(i18n)`Cancel receivable "${receivable.document_id}"?`
+  ) : (
+    t(i18n)`Cancel receivable?`
+  );
+
+  const modalDescription = hasNoCreditNotes
+    ? t(
+        i18n
+      )`The Credit note to this invoice was created earlier. Following that you can’t cancel invoice.`
```

Comment:
I realize it's not in the design, but it would be nice to at least explain to the user what Credit Note they have for the invoice. A simple `document_id` would suffice.